### ISSUE TITLE
[photospline] use of signed int for ndim (Trac #1784)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1776]
static analysis caught to bugs for `photospline/private/lib/bspline.c` both related to considering tables with negative dimensions, probably just needs some `assert(n>0)`

http://software.icecube.wisc.edu/static_analysis/2016-07-21-030215-62891-1/report-40f451.html#EndPath
http://software.icecube.wisc.edu/static_analysis/2016-07-21-030215-62891-1/report-8ba5f2.html#EndPath

Migrated from https://code.icecube.wisc.edu/ticket/1784
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:12:38", 
    "description": "static analysis caught to bugs for `photospline/private/lib/bspline.c` both related to considering tables with negative dimensions, probably just needs some `assert(n>0)`\n\nhttp://software.icecube.wisc.edu/static_analysis/2016-07-21-030215-62891-1/report-40f451.html#EndPath\nhttp://software.icecube.wisc.edu/static_analysis/2016-07-21-030215-62891-1/report-8ba5f2.html#EndPath", 
    "reporter": "kjmeagher", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067158057333", 
    "component": "combo reconstruction", 
    "summary": "[photospline] use of signed int for ndim", 
    "priority": "critical", 
    "keywords": "", 
    "time": "2016-07-21T17:28:55", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160722T16:21:51: 
* olivas changed status from "new" to "assigned"
* olivas changed milestone from "" to "July 2016"
* olivas changed owner from "" to "jvansanten"
* olivas changed component from "cmake" to "icerec"

**Comment by icecube-trac** [comment]
Trac update at 20160729T08:25:14: 
* jvansanten changed status from "assigned" to "closed"
* jvansanten changed resolution from "" to "fixed"
* jvansanten commented:

> In changeset 148888:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="148888"
> Add more annotations to keep static analysis happy. Fixes #1776.
> ```

**Comment by icecube-trac** [comment]
Trac update at 20160805T16:28:39: 
* olivas commented:

> There's a good chance this is related to #1805 and both can likely be closed at the same time when this is resolved.


* olivas changed resolution from "fixed" to ""
* olivas changed status from "closed" to "reopened"
* olivas changed milestone from "July 2016" to "August 2016"
* olivas changed priority from "normal" to "critical"

**Comment by icecube-trac** [comment]
Trac update at 20160815T08:44:14: 
* jvansanten commented:

> In changeset 149245:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="149245"
> Add annotations to keep static analysis happy. Fixes #1805 and #1776.
> 
> Also, add a note to the next person who forgets about the tricks
> employed to avoid special-casing evaluations outside the full support
> of the spline.
> ```


* jvansanten changed status from "reopened" to "closed"
* jvansanten changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:12:38: 
* olivas changed milestone from "August 2016" to ""
* olivas commented:

> Milestone August 2016 deleted
